lib/routes/hfut/hf/utils.ts

```diff
@@ -0,0 +1,87 @@
+import cache from '@/utils/cache';
+import got from '@/utils/got';
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+
+const typeMap = {
+    tzgg: { name: 'tzgg', url: 'https://news.hfut.edu.cn/tzgg2.htm', root: 'https://news.hfut.edu.cn', title: '合肥工业大学 - 通知公告' },
+    jxky: { name: 'jxky', url: 'https://news.hfut.edu.cn/tzgg2.htm', root: 'https://news.hfut.edu.cn', title: '合肥工业大学 - 通知公告 - 教学科研' },
+    qttz: { name: 'qttz', url: 'https://news.hfut.edu.cn/tzgg2.htm', root: 'https://news.hfut.edu.cn', title: '合肥工业大学 - 通知公告 - 其它通知' },
+};
+
+const commLink = 'https://news.hfut.edu.cn/';
+
+const parseList = async (ctx, type) => {
+    const link = typeMap[type].url;
+    const title = typeMap[type].title;
+
+    const response = await got(link);
+    const $ = load(response.data);
+
+    const resultList = await parseArticle(typeMap[type].name, $);
+
+    return {
+        title,
+        link,
+        resultList,
+    };
+};
+
+async function parseArticle(type, $) {
+    let data = $('#tzz').find('li').toArray();
+
+    if (type === 'jxky') {
+        data = $('#c01').find('li').toArray();
+    } else if (type === 'qttz') {
+        data = $('#c02').find('li').toArray();
+    }
+
+    const items = data.map((item) => {
+        item = $(item);
+        const oriLink = item.find('a').attr('href');
+        let linkRes = oriLink;
+        if (!oriLink.includes('http')) {
```

Comment:
The condition will also be `true` if `http` appears in the middle of `oriLink`
```suggestion
        if (!oriLink.startsWith('http')) {
```